zstacklib/zstacklib/utils/gpu.py

```
@@ -996,3 +1015,549 @@ def parse_alibaba_ppu_metric_output(output):
+                        result = gpu_info.to_addon_dict()
+
+                        # Handle vendor-specific extra fields
+                        if vendor_name == VendorEnum.HUAWEI:
+                            # Huawei special handling: npuId, isIsolated already in extra
+                            result.update(gpu_info.extra)
+
+                            # Collect product name and aios rank table (using legacy functions)
+                            try:
+                                npu_ids = plugin.get_npu_ids()  # Class method
+                                if npu_ids:
+                                    r, o, e = bash_roe(
```

Ruff (0.14.13)

`bash_roe` may be undefined, or defined from star imports

(F405)